core/src/main/java/com/yahoo/ycsb/generator/UnixEpochTimestampGenerator.java

```diff
@@ -107,28 +111,35 @@ public void initalizeTimestamp(final long intervalOffset) {
     switch (timeUnits) {
     case NANOSECONDS:
       currentTimestamp = System.nanoTime() + getOffset(intervalOffset);
+      startTimestamp = currentTimestamp;
       break;
     case MICROSECONDS:
       currentTimestamp = (System.nanoTime() / 1000) + getOffset(intervalOffset);
+      startTimestamp = currentTimestamp;
       break;
     case MILLISECONDS:
       currentTimestamp = System.currentTimeMillis() + getOffset(intervalOffset);
+      startTimestamp = currentTimestamp;
       break;
     case SECONDS:
       currentTimestamp = (System.currentTimeMillis() / 1000) +
           getOffset(intervalOffset);
+      startTimestamp = currentTimestamp;
       break;
     case MINUTES:
       currentTimestamp = (System.currentTimeMillis() / 1000) +
           getOffset(intervalOffset);
+      startTimestamp = currentTimestamp;
       break;
     case HOURS:
       currentTimestamp = (System.currentTimeMillis() / 1000) +
           getOffset(intervalOffset);
+      startTimestamp = currentTimestamp;
       break;
     case DAYS:
       currentTimestamp = (System.currentTimeMillis() / 1000) +
           getOffset(intervalOffset);
+      startTimestamp = currentTimestamp;
```

Comment:
nit: could do this `startTimestamp = currentTimestamp` could go right before the return after the switch/case